### PR TITLE
Bugfixes, minor improvement, performance improvement and small cleanup

### PR DESCRIPTION
**Pull request by PatrickvonDeetzen**
### **TL;DR**
- 4 small bugfixes (we encountered 2 every now and then at work in GF while working on our projects and 2 more i looked up from open issues here on GitHub) 
- 1 minor improvement in case a form/class can´t be opened
- Performance improvements to the searchengine
- Some small cleanup to the code 

================================

### **Summary:**

**- Bug 1:** 
On clicking "Edit" to open the file of the selected result could lead to an error "Variable 'CR' is not found.". This occured because in "gf_peme_basetools.prg" the file "GoFish.h" was not included but one of it´s variables was used.
(see file "1 - error message.png")

**- Bug 2:** 
For results that are part of a class-container and used in a form with a formset GF showed a wrong "name" (or you may call it "path") in the results. This was because for handling VCX files there was code that messed up in this special case and changed nothing in every other case. (see other files)

**- Bug 3 (change for Issue #3):** 
When a result set is shown and you click the header of one of the search columns "Proccode", "Statement" or "Code" GoFish will run into an error "SQL: ORDER BY clause in invalid.". This is because those are memo fields and VFP can´t sort by those. 

**- Bug 4 (change for Issue #4 ):** 
When the search option "Whole word" is active then in the HTML preview may be too many highlightings applied in lines with a matching result. This was because the higlighting got applied two times using two different functions. (Besides that the HTML code also got pretty messy.)

**- Minor improvement:** 
When clicking "Edit" to open a form or class of the selected result and this file could not be opened then nothing happened. From now on GF will tell you that the file can't be opened and why not. For example: The class you want to open is already in use by another form you have still opened or the form/class is already used/opened in another instance of VFP (or even another programm).  

**- Performance Improvements:**
a) In procedure "IsFullLineComment" the object used for a RegEx search was always created in a new instance and then closed again. This is now created once in the procedure "init".
b) In procedure "AssignMatchTypeForScxVcx" the order of the checks in the "DO ... CASE ..." got slightly changed due to likelyhood of occurence.
c) In procedure "IsFileTypeIncluded" the existing code was rewritten to be executed in as little steps/cpu cycles as possible and leaving out all the nonsene operations/declarations.
d) In procedure "SearchInFileName" we check for the datetime of the file *after* we checked there is a matching result in this file, because the procedure "GetFileDateTime" might read the whole file with "USE ..." which kind of takes a while to execute. 
e) In procedure "SearchInProject" added commented out code for measuring execution time of the search (aka "performance test").
=> Depending on project size and number of found results the search finished on average 13% to 37% faster with this changes.

**- Small cleanup:** 
a) In procedure "GetFileDateTime" the declared variable "lcFileName" is never used, but "ldFileName" was not declared but used.
b) In procedure "GetFileDateTime" i deleted some old code that was already commented out.
c) In procedure "SetSearchError" i deleted some old code that was already commented out.
d) In procedure "ShowError" i deleted some old code that was already commented out.

==============

Feel free to comment these changes or roast me (if necessary). I hope you all can somewhat understand my english, since i try my best to be clear.

P.S.: Lots of kudos to all the bigbrain creators and maintainers that made GoFish to what it is today. I know i have already learned a lot by digging through some parts of this code. ;-)

-------
1 - error message
2a - wrong results
2b - expected result
2c - expected result
-

### CODE REVIEW

**Comment by PatrickvonDeetzen** [comment]
So far any questions or suggetions from anyone? o.o
